resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/SslConfigurationProvider.scala

```diff
@@ -112,92 +134,197 @@ private[spark] class SslConfigurationProvider(
+          s" a file or does not exist.")
+      }
+      val keyStoreBytes = Files.toByteArray(file)
+      (secretName, BaseEncoding.base64().encode(keyStoreBytes))
+    }).toMap
+  }
+
+  private def parseDriverSubmitSslOptions(): SubmissionSSLOptions = {
+    val maybeKeyStore = sparkConf.get(DRIVER_SUBMIT_SSL_KEYSTORE)
+    val maybeTrustStore = sparkConf.get(DRIVER_SUBMIT_SSL_TRUSTSTORE)
+    val maybeKeyPem = sparkConf.get(DRIVER_SUBMIT_SSL_KEY_PEM)
+    val maybeServerCertPem = sparkConf.get(DRIVER_SUBMIT_SSL_SERVER_CERT_PEM)
+    val maybeClientCertPem = sparkConf.get(DRIVER_SUBMIT_SSL_CLIENT_CERT_PEM)
+    validatePemsDoNotConflictWithStores(
+      maybeKeyStore,
+      maybeTrustStore,
+      maybeKeyPem,
+      maybeServerCertPem,
+      maybeClientCertPem)
+    val resolvedSparkConf = sparkConf.clone()
+    val (isLocalKeyStore, resolvedKeyStore) = resolveLocalFile(maybeKeyStore, "keyStore")
     resolvedKeyStore.foreach {
-      resolvedSparkConf.set(KUBERNETES_DRIVER_SUBMIT_KEYSTORE, _)
+      resolvedSparkConf.set(DRIVER_SUBMIT_SSL_KEYSTORE, _)
     }
-    sparkConf.get(KUBERNETES_DRIVER_SUBMIT_TRUSTSTORE).foreach { trustStore =>
-      val trustStoreURI = Utils.resolveURI(trustStore)
-      trustStoreURI.getScheme match {
-        case "file" | null =>
-          resolvedSparkConf.set(KUBERNETES_DRIVER_SUBMIT_TRUSTSTORE, trustStoreURI.getPath)
-        case _ => throw new SparkException(s"Invalid trustStore URI $trustStore; trustStore URI" +
-          " for submit server must have no scheme, or scheme file://")
-      }
+    val (isLocalServerCertPem, resolvedServerCertPem): (Boolean, Option[String]) =
+      resolveLocalFile(maybeServerCertPem, "server cert PEM")
+    val (isLocalKeyPem, resolvedKeyPem) = resolveLocalFile(maybeKeyPem, "key PEM")
+    maybeTrustStore.foreach { trustStore =>
+      require(KubernetesFileUtils.isUriLocalFile(trustStore), s"Invalid trustStore URI" +
+        s"$trustStore; trustStore URI for submit server must have no scheme, or scheme file://")
+      resolvedSparkConf.set(DRIVER_SUBMIT_SSL_TRUSTSTORE, Utils.resolveURI(trustStore).getPath)
+    }
+    val clientCertPem: Option[String] = maybeClientCertPem.map { clientCert =>
+      require(KubernetesFileUtils.isUriLocalFile(clientCert), "Invalid client certificate PEM URI" +
+        s" $clientCert: client certificate URI must have no scheme, or scheme file://")
+      Utils.resolveURI(clientCert).getPath
     }
     val securityManager = new SparkSecurityManager(resolvedSparkConf)
-    (securityManager.getSSLOptions(KUBERNETES_SUBMIT_SSL_NAMESPACE), isLocalKeyStore)
+    val storeBasedSslOptions = securityManager.getSSLOptions(DRIVER_SUBMIT_SSL_NAMESPACE)
+    SubmissionSSLOptions(
+      storeBasedSslOptions,
+      isLocalKeyStore,
+      resolvedKeyPem.map(new File(_)),
+      isLocalKeyPem,
+      resolvedServerCertPem.map(new File(_)),
+      isLocalServerCertPem,
+      clientCertPem.map(new File(_)))
```

Comment:
My understanding is that `File` objects aren't opened and closed resources in Java. Actually opening and closing the File should be delegated to classes like `FileIinputStream`.